sympy/core/expr.py

```diff
@@ -300,9 +300,6 @@ def __ge__(self, other):
         if n2 is not None:
             return _sympify(n2 >= 0)
         if self.is_extended_real and other.is_extended_real:
-            if self.is_infinite and other.is_infinite:
-                if (self * other).is_extended_positive:  # inf < inf & -inf < -inf
-                    return S.true
             if (self.is_infinite and self.is_extended_nonnegative) \
```

Comment:
Is this better than `self.is_infinite and self > 0`, or maybe `self is S.Infinity`?

Reply:
we can't use `self is S.Infinity` since we need it to work for symbols with `positive and infinite` assumptions. And though `self >= 0` can be used here, I believe it is better left this way.

Reply:
I have doubts about the necessity or usefulness  of "extended positive". I would rather postpone its definition to a time when the comparison system has stabilized.

Reply:
I doubt using `self >= 0` or etc, may call the same functions `Expr.__lt__` and others.

Reply:
That is possible. I have not studied the dependencies of the assumption system.

Reply:
It is the concept of "extended positive" that seems artificial to me. I hope that the assumptions code could be written without making use of that internally.

Reply:
This code is for evaluating the relations <, <=, >=, > where possible. It is built on top of the assumptions system. Using self > 0 punts the decision over to `__gt__` which also ultimately has to use `self.is_extended_positive`.

The concept of x being extended positive is precisely the same as x > 0. In the assumptions system it is spelled as x.is_extended_positive. The code in these functions makes that connection between by using assumptions to evaluate relations.

In general for why we need extended positive consider the relationship between the assumptions on x and those on 1/x:
```
1/positive -> positive
1/nonnegative -> extended_positive
1/extended_nonnegative -> extended_nonnegative
```
If x is real then `1/x**2 > 0` but there's no way to conclude that from the assumption system without having is_extended_positive.

Reply:
I understand @jksuom's point here. And, yeah, we can write code in assumptions without using properties like `extended_positive` but logic is better simplified this way and it will make debugging the code easier. Also, as @oscarbenjamin pointed we need some notion of `extended positive` anyways in the assumptions system.

Reply:
I don't think it is possible to make everything work without extended_positive. Note that before #16666 there was still extended_positive: it was just called "positive".

Reply:
It seems to me that the job of `satisfiable` will grow exponentially with the number of different basic predicates. So I was hoping that "extended positive" could be replaced by "positive or oo".

Reply:
In ternary logic it's equivalently possible to combine basic predicates using `and` but not `or`. In the case of `1/x**2` for real x neither of `positive` or `oo` can be definitely true independently but the combined `positive or oo` (is_extended_positive) is known to be true.

Reply:
> It seems to me that the job of satisfiable will grow exponentially with the number of different basic predicates.

Perhaps for the new assumptions `extended_positive` should be automatically replaced with `positive or oo` before being passed to satisfiable. Although I would expect it to be able to do this automatically if the relevant fact is given to it, but it could help performance.

Reply:
This would be clearer as `extended_positive` instead of `extended_nonnegative`

Reply:
The assumptions for a `+oo` value would be `extended_nonnegative` and `infinite`. `extended_positive` is covered. But it can't be done in the other way round.

Reply:
I'm not following what you're saying here.

Reply:
Ok, a variable with assumptions `infinite` and `extended_nonnegative` will be a case of `+oo`, if we `extended_positive` here, such case will be missed.

Reply:
@ShubhamKJha I don't think that's right. In the assumptions we have
```
zero -> finite
infinite -> !finite
```
So anything that is infinite cannot be zero. If zero is excluded as a possibility then extended_positive and extended_nonnegative are equivalent:
```julia
In [1]: x = Symbol('x', infinite=True, extended_nonnegative=True)

In [2]: x.is_extended_positive
Out[2]: True
```

Reply:
Don't we have (extended) nonnegative <-> zero or (extended) positive? In other words, the only difference between nonnegative and positive (extended or no) is zero, which is irrelevant if we also require infinite.

Reply:
@asmeurer that's correct. That's what I was trying to say above.

Reply:
Okay, perhaps I missed that point. If it makes the code clearer I would change it.